### PR TITLE
[PR1 DEV-52 MVP-3

### PR DESCRIPTION
**Pull request by jmaralo**
Opening as Draft because we still need to test orders and messages.

This is the first "full" version of the gui, it can fully communicate the front with the pod, but it still needs optimizations and improvements.

### CODE REVIEW

File: src/Shared/Server/infra/mappers/orders.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	"strconv"
+
+	excelAdapter "github.com/HyperloopUPV-H8/Backend-H8/Shared/excel_adapter/domain"
+	"github.com/HyperloopUPV-H8/Backend-H8/Shared/server/domain"
+)
+
+func GetOrders(boards map[string]excelAdapter.BoardDTO) map[uint16]domain.Order {
+	orders := make(map[uint16]domain.Order)
+	for _, board := range boards {
+		for _, packet := range board.GetPackets() {
```
Comment (smorsua) [inline src/Shared/Server/infra/mappers/orders.go:14]: Check if packet is Order type

File: src/Shared/Server/domain/board.go [hunk 2]
```diff
@@ -1,7 +1,6 @@
 package domain
 
 type Board struct {
-	Name    string            `json:"name"`
-	Packets map[uint16]Packet `json:"packets"`
-	Orders  map[uint16]Packet `json:"orders"`
```
Comment (smorsua) [inline src/Shared/Server/domain/board.go:6]: Change back to maps

File: src/Shared/Server/domain/pod_data.go [hunk 3]
```diff
@@ -1,5 +1,6 @@
 package domain
 
 type PodData struct {
-	Boards map[string]Board `json:"boards"`
```
Comment (smorsua) [inline src/Shared/Server/domain/pod_data.go:4]: Change back to map

File: src/Shared/Server/infra/mappers/packet.go [hunk 4]
```diff
@@ -8,24 +8,13 @@ import (
 	"github.com/HyperloopUPV-H8/Backend-H8/Shared/server/domain"
 )
 
-func getPackets(packets []excelAdapter.PacketDTO) map[uint16]domain.Packet {
-	result := make(map[uint16]domain.Packet, len(packets))
+func getPackets(packets []excelAdapter.PacketDTO) []domain.Packet {
+	result := make([]domain.Packet, 0, len(packets))
 	for _, packet := range packets {
-		if packet.Description.Direction == "Input" {
```
Comment (smorsua) [inline src/Shared/Server/infra/mappers/packet.go:14]: Check == Data instead of != Input

File: src/Shared/Server/infra/pod_data.go [hunk 5]
```diff
@@ -2,14 +2,15 @@ package infra
...
 	"encoding/json"
+	"fmt"
 	"log"
 	"net/http"
-
-	"github.com/HyperloopUPV-H8/Backend-H8/Shared/server/domain"
 )
 
-func (server HTTPServer[D, O, M]) HandlePodData(route string, podData domain.PodData) {
+func (server HTTPServer[D, O, M]) HandlePodData(route string, podData any) {
```
Comment (smorsua) [inline src/Shared/Server/infra/pod_data.go:10]: This should be changed in the next version

File: src/cmd/MVP-3/main.go [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+	orderTransferPresentation "github.com/HyperloopUPV-H8/Backend-H8/order_transfer/infra/presentation"
+	"github.com/joho/godotenv"
+)
+
+var cpuprofile = flag.String("cpuprofile", "", "write cpu profile to file")
+
+func main() {
+	flag.Parse()
+	if *cpuprofile != "" {
+		f, err := os.Create(*cpuprofile)
```
Comment (smorsua) [inline src/cmd/MVP-3/main.go:38]: Put inside function

File: src/cmd/MVP-3/main.go [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+		if err != nil {
+			log.Fatal(err)
+		}
+		pprof.StartCPUProfile(f)
+		defer pprof.StopCPUProfile()
+	}
+
```
Comment (smorsua) [inline src/cmd/MVP-3/main.go:45]: Put each log + function call inside a funtion to be more concise in main

File: src/cmd/MVP-3/main.go [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+	log.Println("\tDone!")
+
+	log.Println("Setting up server...")
+	server := server.New[
+		dataTransferApplication.PacketJSON,
+		orderTransferApplication.OrderJSON,
+		messageTransferApplication.MessageJSON]()
+
+	log.Println("\t\tStarting data routine")
+	go func() {
```
Comment (smorsua) [inline src/cmd/MVP-3/main.go:85]: Put in function

File: src/cmd/MVP-3/main.go [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+		}
+	}()
+	server.HandleWebSocketOrder("/backend/"+os.Getenv("ORDER_ENDPOINT"), orderTransferPresentation.OrderRoutine)
+	log.Println("\t\t\tDone!")
+
+	log.Println("\t\tStarting message routine")
+	go func() {
+
+	}()
+	server.HandleWebSocketMessage("/backend/"+os.Getenv("MESSAGE_ENDPOINT"), messageTransferPresentation.MessageRoutine)
```
Comment (smorsua) [inline src/cmd/MVP-3/main.go:122]: Put all Handle... in a single function

**Review by smorsua — COMMENTED** [review]
Look pretty good overall, let me know when you update the PR

File: src/Shared/Server/domain/board.go (hunk 2 above)
Comment (lipezaballa) [reply src/Shared/Server/domain/board.go:6]: Why the change had been made?

File: src/Shared/Server/domain/measurement.go [hunk 10]
```diff
@@ -1,13 +1,8 @@
 package domain
 
 type Measurement struct {
-	Name   string `json:"Name"`
-	Value  string `json:"Value"`
-	Ranges Ranges `json:"ranges"`
-	Units  string `json:"units"`
-}
-
-type Ranges struct {
```
Comment (lipezaballa) [inline src/Shared/Server/domain/measurement.go:10]: how we manage the ranges now?

File: src/Shared/Server/infra/mappers/measurement.go [hunk 11]
```diff
@@ -1,48 +1,37 @@
...
-		log.Fatalln(err)
-	}
-	end, err := strconv.ParseFloat(strings.Trim(parts[0], " "), 64)
-	if err != nil {
-		log.Fatalln(err)
-	}
-
-	return [2]float64{start, end}
+func getType(kind string) string {
+	return "Number"
```
Comment (lipezaballa) [inline src/Shared/Server/infra/mappers/measurement.go:36]: why always "Number"?

**Comment by jmaralo** [comment]
Closing because it's obsoleted after the structure refactor

**Pull request closed without merging**